stock_inventory_revaluation/models/stock_inventory_revaluation.py

```diff
@@ -0,0 +1,273 @@
+                "move_type": "entry",
+            }
+            valuation_layer_ids = []
+            revaluation_to_add_byproduct = defaultdict(lambda: 0.0)
+            for line in revaluation.stock_inventory_revaluation_line_ids:
+                remaining_qty = sum(
+                    line.stock_move_id.stock_valuation_layer_ids.mapped("remaining_qty")
+                )
+                if line.stock_move_id._is_out():
+                    remaining_qty = -1 * sum(
+                        line.stock_move_id.stock_valuation_layer_ids.mapped("quantity")
+                    )
+                linked_layer = line.stock_move_id.stock_valuation_layer_ids[:1]
+                # Prorate the value at what's still in stock
+                revaluation_to_add = (
+                    remaining_qty / line.stock_move_id.product_qty
+                ) * line.additional_value
+                if not revaluation.company_id.currency_id.is_zero(revaluation_to_add):
+                    valuation_layer = self.env["stock.valuation.layer"].create(
+                        {
+                            "value": revaluation_to_add,
+                            "quantity": 0,
+                            "remaining_qty": 0,
+                            "stock_valuation_layer_id": linked_layer.id,
+                            "description": revaluation.name,
+                            "stock_move_id": line.stock_move_id.id,
+                            "product_id": line.product_id.id,
+                            # The layer is linked to the revaluation line
+                            "stock_inventory_revaluation_line_id": line.id,
+                            "company_id": revaluation.company_id.id,
+                        }
+                    )
+                    line.created_stock_valuation_layer_id = valuation_layer.id
+                    linked_layer.remaining_value += revaluation_to_add
+                    valuation_layer_ids.append(valuation_layer.id)
+                # Update the AVCO
+                product = line.product_id
+                if product.cost_method == "average":
+                    revaluation_to_add_byproduct[product] += revaluation_to_add
+                # Products with manual inventory valuation are ignored because they
+                # do not need to create journal entries.
+                if product.valuation != "real_time":
+                    continue
+                # `remaining_qty` is negative if the move is out and delivered proudcts
+                #  that were not
+                # in stock.
+                qty_out = 0
+                if line.stock_move_id._is_in():
+                    qty_out = line.stock_move_id.product_qty - remaining_qty
+                elif line.stock_move_id._is_out():
+                    qty_out = line.stock_move_id.product_qty
+                move_vals["line_ids"] += line._create_accounting_entries(
+                    move, line, qty_out
+                )
```

Comment:
if additional_value is zero avoid create account.move.line 

![image](https://user-images.githubusercontent.com/7775116/156858911-b9da9565-d347-4c5d-a54c-a62cc5fd5729.png)

![image](https://user-images.githubusercontent.com/7775116/156858929-cf8e2224-cf2b-4339-a474-9b6a530e7f7d.png)

Reply:
I include a filtered in the method for that purpose, it should work now.